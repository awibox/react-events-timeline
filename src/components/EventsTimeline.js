const EventsTimeline = () => {};

export default EventsTimeline;
