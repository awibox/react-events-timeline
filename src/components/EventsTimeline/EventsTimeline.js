import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './EventsTimeline.scss';

const EventsTimeline = (props) => {
  const { title, icon, data } = props;
  return (
    <div className={classNames(styles.timeline, {
      [styles[props.color]]: props.color,
    })}>
      <div className={styles.title}>
        {icon && <div className={styles.icon}>
          <i className={`fa fa-${icon}`} />
        </div>
        }
        <span>{title}</span>
      </div>
      {
        data && data.map((item) => (
          <div key={item.year} className={styles.event}>
            <div className={styles.year}>{item.year}</div>
            <div className={styles.content}>
              {item.content}
            </div>
          </div>
        ))
      }
    </div>
  );
};

EventsTimeline.propTypes = {
  title: PropTypes.string,
  icon: PropTypes.string,
  color: PropTypes.string,
  data: PropTypes.any,
};

export default EventsTimeline;
