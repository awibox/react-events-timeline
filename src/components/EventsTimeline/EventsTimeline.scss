$orange: #fba51a;
$blue: #0079c1;
$green: #40bea9;
$widthScreenMobileLarge: 480px;
$timelineIndent: 10px;
$yearRadius: 50px;
$iconRadius: 60px;

.timeline {
  background: #f3f3f4;
  position: relative;
  &:before {
    content: '';
    position: absolute;
    top: 10px;
    left: 34px;
    height: 100%;
    width: 4px;
    background: #e7eaec;
  }
  &.green {
    .title i {
      color: $green;
    }
    .year {
      background: $green;
    }
  }
  &.blue {
    .title i {
      color: $blue;
    }
    .year {
      background: $blue;
    }
  }
  &.orange {
    .title i {
      color: $orange;
    }
    .year {
      background: $orange;
    }
  }
}
.title {
  min-height: 40px;
  display: flex;
  flex-flow: row;
  align-items: center;
  position: relative;
  z-index: 3;
  margin: 0 1px;
  padding-bottom: 10px;
  span {
    padding: 10px 0;
    font-size: 22px;
    margin: 0 20px;
  }
}
.icon {
  width: $iconRadius;
  background-color: #fff;
  min-width: $iconRadius;
  height: $iconRadius;
  min-height: $iconRadius;
  text-align: center;
  line-height: $iconRadius;
  font-size: 24px;
  border-radius: 50%;
  border: 4px solid #e7eaec;
}
.event {
  width: 100%;
  position: relative;
  display: flex;
  flex-flow: row;
  z-index: 2;
}
.year {
  width: $yearRadius;
  min-width: $yearRadius;
  height: $yearRadius;
  min-height: $yearRadius;
  text-align: center;
  line-height: $yearRadius;
  background: #333;
  color: #fff;
  border-radius: 50%;
  margin: $timelineIndent;
}
.content {
  position: relative;
  margin: $timelineIndent;
  width: 100%;
  background: white;
  border-radius: $timelineIndent/4;
  padding: $timelineIndent*2;
  margin-right: 0;
  box-shadow: 1px 1px 2px rgba(0, 0, 0, 0.12);
  &:before {
    content: '';
    position: absolute;
    top: $yearRadius/3;
    right: 100%;
    height: 0;
    width: 0;
    border: 7px solid transparent;
    border-right: 7px solid white;
  }
}
@media (max-width: $widthScreenMobileLarge) {
  .timeline {
    &:before {
      left:50%;
      margin-left: -2px;
      top:50px;
    }
  }
  .title {
    text-align: center;
    flex-flow: column;
    span {
      display: none;
    }
  }
  .event {
    flex-flow: column-reverse;
    align-items: center;
    .content {
      width: auto;
      margin: 10px 0;
      &:before {
        display: none;
      }
    }
  }
}
