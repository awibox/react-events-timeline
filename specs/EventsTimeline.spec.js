import React from 'react';
import { shallow } from 'enzyme';
import EventsTimeline from '../src/components/EventsTimeline/EventsTimeline';

describe('EventsTimeline', () => {
  it('should render correctly', () => {
    const component = shallow(<EventsTimeline />);
    expect(component).toMatchSnapshot();
  });
});
