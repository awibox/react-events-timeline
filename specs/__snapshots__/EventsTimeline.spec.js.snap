// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsTimeline should render correctly 1`] = `
<div
  className="timeline"
>
  <div
    className="title"
  >
    <span />
  </div>
</div>
`;
